index.d.ts

```diff
@@ -2912,16 +2912,14 @@ export type DeleteRecordOperationGo<ResponseType, Keys> = <
   options?: Options,
 ) => Options extends infer O
   ? "response" extends keyof O
-    ? O["response"] extends "all_new"
-      ? Promise<{ data: T }>
-      : O["response"] extends "all_old"
-      ? Promise<{ data: T }>
+    ? O["response"] extends "all_old"
+      ? Promise<{ data: T | null }>
       : O["response"] extends "default"
-      ? Promise<{ data: Keys }>
+      ? Promise<{ data: Keys | null }>
       : O["response"] extends "none"
       ? Promise<{ data: null }>
-      : Promise<{ data: Partial<T> }>
-    : Promise<{ data: Keys }>
+      : Promise<{ data: Keys | null }>
```

Comment:
@ikeyan thank you for the PR!

Have you experienced scenarios where you receive back a `null` value from `delete`? If so, could you help me recreate them?

Reply:
@tywalch sure, `null` is returned when there is no matching item to return.

Reply:
Merging but that won't publish automatically (I'll try to do that tonight) 

Thank you so much for this 💪